lib/matplotlib/_constrained_layout.py

```diff
@@ -276,21 +277,35 @@ def _make_layout_margins(fig, renderer, *, w_pad=0, h_pad=0,
 def _make_margin_suptitles(fig, renderer, *, w_pad=0, h_pad=0):
     # Figure out how large the suptitle is and make the
     # top level figure margin larger.
+
+    invTransFig = fig.transFigure.inverted().transform_bbox
+    # get the h_pad and w_pad as distances in the local subfigure coordinates:
+    padbox = mtransforms.Bbox([[0, 0], [w_pad, h_pad]])
+    padbox = (fig.transFigure -
+                   fig.transSubfigure).transform_bbox(padbox)
+    h_pad_local = padbox.height
+    w_pad_local = padbox.width
+
     for panel in fig.subfigs:
         _make_margin_suptitles(panel, renderer, w_pad=w_pad, h_pad=h_pad)
 
     if fig._suptitle is not None and fig._suptitle.get_in_layout():
-        invTransFig = fig.transSubfigure.inverted().transform_bbox
-        parenttrans = fig.transFigure
-        w_pad, h_pad = (fig.transSubfigure -
-                        parenttrans).transform((w_pad, 1 - h_pad))
-        w_pad, one = (fig.transSubfigure -
-                      parenttrans).transform((w_pad, 1))
-        h_pad = one - h_pad
-        bbox = invTransFig(fig._suptitle.get_tightbbox(renderer))
         p = fig._suptitle.get_position()
-        fig._suptitle.set_position((p[0], 1-h_pad))
-        fig._layoutgrid.edit_margin_min('top', bbox.height + 2 * h_pad)
+        fig._suptitle.set_position((p[0], 1 - h_pad_local))
+        bbox = invTransFig(fig._suptitle.get_tightbbox(renderer))  # fig coords
+        fig._layoutgrid.edit_margin_min('top', bbox.height + 2.0 * h_pad)
```

Comment:
Small inconsistency.
```suggestion
        fig._layoutgrid.edit_margin_min('top', bbox.height + 2 * h_pad)
```

Reply:
I went the other way.  I know that non-floats are fashionable in python, but I've *really* been stumped a few times when an array went all integers, so I strongly prefer being explicit.

Reply:
There's still one on line 308, then